x-pack/plugins/siem/public/timelines/components/timeline/body/events/event_column_view.tsx

```diff
@@ -79,17 +86,70 @@ export const EventColumnView = React.memo<Props>(
     toggleShowNotes,
     updateNote,
   }) => {
-    const timelineTypeContext = useTimelineTypeContext();
-
+    const { getManageTimelineById } = useManageTimeline();
+    const timelineActions = useMemo(() => getManageTimelineById(timelineId).timelineRowActions, [
+      getManageTimelineById,
+      timelineId,
+    ]);
     const additionalActions = useMemo<JSX.Element[]>(() => {
-      return (
-        timelineTypeContext.timelineActions?.map((action) => (
-          <EventsTdContent key={action.id} textAlign="center">
-            {action.getAction({ eventId: id, ecsData })}
-          </EventsTdContent>
-        )) ?? []
+      const grouped = timelineActions.reduce(
+        (
+          acc: {
+            contextMenu: JSX.Element[];
+            icon: JSX.Element[];
+          },
+          action
+        ) => {
+          if (action.displayType === 'icon') {
+            return {
+              ...acc,
+              icon: [
+                ...acc.icon,
+                <EventsTdContent key={action.id} textAlign="center">
+                  <EuiToolTip
+                    data-test-subj={`${action.dataTestSubj}-tool-tip`}
+                    content={action.content}
+                  >
+                    <EuiButtonIcon
+                      aria-label={action.ariaLabel}
+                      data-test-subj={`${action.dataTestSubj}-button`}
+                      iconType={action.iconType}
+                      isDisabled={action.isActionDisabled ?? false}
+                      onClick={() => action.onClick({ eventId: id, ecsData })}
+                    />
+                  </EuiToolTip>
+                </EventsTdContent>,
+              ],
+            };
+          }
+          return {
+            ...acc,
+            contextMenu: [
+              ...acc.contextMenu,
+              <EuiContextMenuItem
+                aria-label={action.ariaLabel}
+                data-test-subj={action.dataTestSubj}
+                disabled={action.isActionDisabled ?? false}
+                icon={action.iconType}
+                onClick={() => action.onClick({ eventId: id, ecsData })}
```

Comment:
Can we wrap the `onClick` actions in a function that closes the parent `EuiPopover` once clicked? Currently the action must complete before the popover is closed:

<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/2946766/83461175-3ea57f00-a425-11ea-930b-7a99e60d9382.gif" />
</p>